kubectl-plugin/pkg/cmd/submit/submit.go

```diff
@@ -0,0 +1,357 @@
+	cmd.Flags().StringVar(&options.workingDir, "working-dir", options.workingDir, "Directory containing files that your job will run in")
+	cmd.Flags().Float32Var(&options.entryPointCPU, "entrypoint-num-cpus", options.entryPointCPU, "Number of CPU reserved for the for the entrypoint command")
+	cmd.Flags().Float32Var(&options.entryPointGPU, "entrypoint-num-gpus", options.entryPointGPU, "Number of GPU reserved for the for the entrypoint command")
+	cmd.Flags().IntVar(&options.entryPointMemory, "entrypoint-memory", options.entryPointMemory, "Amount of memory reserved for the entrypoint command")
+	cmd.Flags().BoolVar(&options.noWait, "no-wait", options.noWait, "If present, will not stream logs and wait for job to finish")
+	options.configFlags.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (options *SubmitJobOptions) Complete() error {
+	if *options.configFlags.Namespace == "" {
+		*options.configFlags.Namespace = "default"
+	}
+
+	if options.workingDir != "" {
+		if options.fileName == "" {
+			return fmt.Errorf("Ray Job file -f/--filename is required when using --working-dir")
+		}
+	}
+
+	if !strings.HasPrefix(options.address, "http://") {
+		options.address = "http://" + options.address
+	}
+
+	options.runtimeEnv = filepath.Clean(options.runtimeEnv)
+	return nil
+}
+
+func (options *SubmitJobOptions) Validate() error {
+	// Overrides and binds the kube config then retrieves the merged result
+	config, err := options.configFlags.ToRawKubeConfigLoader().RawConfig()
+	if err != nil {
+		return fmt.Errorf("Error retrieving raw config: %w", err)
+	}
+	if len(config.CurrentContext) == 0 {
+		return fmt.Errorf("no context is currently set, use %q to select a new one", "kubectl config use-context <context>")
+	}
+
+	if len(options.workingDir) > 0 {
+		info, err := os.Stat(options.workingDir)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("Working dir path does not exist. Failed with: %w", err)
+		} else if err != nil {
+			return fmt.Errorf("Error occurred when checking working dir file: %w", err)
+		} else if !info.IsDir() {
+			return fmt.Errorf("Working dir is not a directory, please use directory: %w", err)
+		}
+
+		// RayJob yaml validation
+		info, err = os.Stat(options.fileName)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("Ray Job file does not exist. Failed with: %w", err)
+		} else if err != nil {
+			return fmt.Errorf("Error occurred when checking ray job file: %w", err)
+		} else if !info.Mode().IsRegular() {
+			return fmt.Errorf("Filename given is not a regular file. Failed with: %w", err)
+		}
+
+		err = decodeRayJobYaml(options.fileName, options.RayJob)
+		if err != nil {
+			return fmt.Errorf("Failed to decode RayJob Yaml: %w", err)
+		}
+
+		submissionMode := options.RayJob.Object["spec"].(map[string]interface{})["submissionMode"]
+		if submissionMode != nil {
+			if submissionMode == rayv1api.K8sJobMode || submissionMode == rayv1api.HTTPMode {
```

Comment:
Probably safer to make this if `submissionMode != rayv1api.UserMode` in case we add more submission modes in the future

Reply:
Currently using `"UserMode"` string since the latest release does not have `rayv1api.UserMode`. Also added a check that returns an error if the ray job does not have submission mode since I did not see https://github.com/ray-project/kuberay/pull/2364/ doing defaulting if submission mode is none.